mars/dataframe/arithmetic/add.py

```diff
@@ -30,18 +30,25 @@ class DataFrameAdd(DataFrameOperand, DataFrameBinOpMixin):
     _fill_value = Float64Field('fill_value')
     _lhs = AnyField('lhs')
     _rhs = AnyField('rhs')
-    _func_name = 'add'
-
-    def __init__(self, axis=None, level=None, fill_value=None, object_type=None, lhs=None, rhs=None, **kw):
-        super(DataFrameAdd, self).__init__(_axis=axis, _level=level,
+    # NB: `_func_name` cannot be defined as a class property for operand op classes,
+    # since the `op` and `rop` may not symmetric.
+    #
+    # For example, we can run df.div(series, axis='columns') but we cannot run series.rdiv(df, axis='columns')
```

Comment:
I cannot see any method to achieve this without and serialized `_func_name`. The `df + series` cases will use `DataFrame.add` rather than be resolved as `Series.radd`, that's fine, but when user explicitly invoke `series.radd(df)` we seem have no way to overload `_rcall` properly...

Reply:
Maybe we just check the type in the rcall for DataFrameDivide? if the lhs is Series and rhs is DataFrame, raise some error.

Reply:
> if the lhs is Series and rhs is DataFrame, raise some error.

But pandas works well for such cases.

Reply:
I finally get it done by adding a `_func_name` property and do some _smart_ resolution on arguments.